### ISSUE TITLE
Library Campaigns and Campaign Descriptions

### ISSUE THREAD

**Issue by fjbagley** [bytedeck/bytedeck#1710]
With library progressing, perhaps the best way for people to see what is available would be  through campaign view? Eventually there will be lots of quests in the library and server seems to slow the more quests there are in quest view.

Would be great to have campaign descriptions so users could get an overview of what the campaign is about.

**Comment by tylerecouture** [comment]
Description should go here, in detail view, just like it does for quests

![Image](https://github.com/user-attachments/assets/22922939-50f9-4760-9af4-deeabce0ec37)

Where should it go in the Library list though? Since currently can't get a detailed view?  Maybe we need to create a detailed view for Campaigns in the library....

**Comment by fjbagley** [comment]
Maybe accordion on click like in the quest.?

**Comment by tylerecouture** [comment]
Ideally it matches the local campaign view, so yes, accordions with additional details for the quest list would be good.  However the buttons will need to be different, I think there would just be View details (to view the full quest) and an import button.

Currently the Quests Library Tab accordion is broken too.

**Comment by thefoxwithin** [comment]
@tylerecouture
I've added library versions of the bootstrap-table-accordion.js and css files and made ajax views for the library quest info and now the right campaign name and Icons are displayed in the library tab and the accordion dropdown works

![Image](https://github.com/user-attachments/assets/37e81add-49d6-493d-af2d-b5354212701e)

Here's a screenshot of the library page for reference

![Image](https://github.com/user-attachments/assets/95a5bae6-720e-451a-badc-b1a9d4f83318)

Notice how the quests are showing the right icons, campaigns and tags even. The ajax_quest_info is sending this page all the right information from the Library schema, so our need for campaign import_ids is lessened. While writing this I noticed that there are four quests not being loaded, I'll address that before I move on to the `View Details`.

I've done my best to keep it as consistent as possible, let me know if there's any other information you want in the dropdown. Right now it just has the icon quest description and the buttons. Currently the `View Details` button doesn't go anywhere because I haven't created the view yet.

I think I'll work on the `View Details`, for the quests so the button can lead somewhere, next. For the Campaigns list do you want it to have the accordion drop down and `View Details`  button or just the `View Details`

Currently this is what the Campaigns list looks like.

![Image](https://github.com/user-attachments/assets/9ced762c-5dc7-4d72-935d-e7ab35f0d6e6)

**Comment by thefoxwithin** [comment]
#  Progress Update – @tylerecouture

I updated the **Library** tab so that it now displays all active quests


. I also added a quest count next to the **Library** tab to match the behavior in other tabs. If we want to show counts for all tabs (e.g., Drafts, Active, etc.), that should be handled in a separate issue—I’ll go ahead and create one after this.

While working on `quest_detail.html`, I realized the template would be duplicated in `confirm_import_quest.html`, so I paused work on that to avoid redundancy.

I also fixed the `status_icons` column—it now displays correctly. I have some additional changes staged on another branch for [issue #1794](https://github.com/bytedeck/bytedeck/issues/1794), but those depend on the accordion logic from the `bootstrap-table-accordion.js` and `.css` that I've implemented here.

---

 **One note**: quest icons in the Library tab don’t always match how they’ll look after import. For example, quests with the “L” image show the Library logo, but that won’t reflect the imported version unless the current deck uses the same logo. I can update this behavior to preview how the icon will look after import if that’s preferred—just let me know.

---

![Image](https://github.com/user-attachments/assets/c435f9ee-734e-4af2-bf32-4f8fa6d641f1)

**Before I open the PR, I’d love your feedback**:

- Anything you'd like to see added to the accordion dropdown?
- Any changes needed on the Campaign Library tab?
- Any other visual tweaks or UI suggestions?

Let me know if you'd like any adjustments!

**Comment by tylerecouture** [comment]
@thefoxwithin I'm a bit confused what this issue has to do with the quest views?  This issue is about campaign descriptions and providing the campaign description from the library campaigns.

**Comment by tylerecouture** [comment]
For the campaign library view, what I want is an info icon that leads to a Library Campaign detail view that matches the non-library campaign detail view:

![Image](https://github.com/user-attachments/assets/6503ad3b-c39f-4621-86de-010dcd84f3e3)

**Comment by tylerecouture** [comment]
> [@tylerecouture](https://github.com/tylerecouture) I've added library versions of the bootstrap-table-accordion.js and css files and made ajax views for the library quest info and now the right campaign name and Icons are displayed in the library tab and the accordion dropdown works

Is this duplication necessary?  Can they not both be handled with the same files by adding some internal logic?  This seems like an unecessary duplication.

**Comment by thefoxwithin** [comment]
> Is this duplication necessary? Can they not both be handled with the same files by adding some internal logic? This seems like an unecessary duplication.

Yeah, I get that—it definitely looks like duplication at first glance. But in this case I think it's worth it, and here's why:

The Library tab behaves pretty differently from the regular deck views. It pulls quests from a separate schema, loads details via AJAX differently, and doesn’t follow the same visibility or deck-based logic. If I tried to cram all of that into the existing accordion JS, I’d need to add a bunch of conditionals and special cases that would make the original code harder to follow and more fragile. 

The main difference is that the rest of the tabs use integer `id`s for quests, while the Library tab references quests by their `import_id` (UUID). That affects how rows are identified, how AJAX URLs are built, and how selectors are matched in the DOM. Trying to support both formats in a single file would’ve introduced a lot of branching logic, so splitting it out made it cleaner and safer.

Having a separate `bootstrap-table-accordion.js` just for the Library means:
- I can tweak and test Library-specific behavior without risking regressions in other tabs.
- It keeps the logic cleaner and more maintainable long term.
- And if the Library evolves—like supporting search filters, inline previews, or more metadata—it’ll be easier to build on this without touching unrelated code.

Same deal with the CSS—`bootstrap-table-accordion.css` for the Library lets us:
- Isolate any visual differences or fixes without accidentally affecting the main deck views.
- Avoid clashing with styles that assume a certain structure or ID format.
- Keep things modular in case the Library UI keeps diverging from the main one.

So yeah, it’s technically some duplication now, but it’s more like intentional separation to prevent future headaches. Happy to revisit if things start overlapping more later!

**Comment by tylerecouture** [comment]
Yes, we definitely want a detail view for the campaigns, and it should look similar or identical to the normal detail view for campaigns!

**Comment by thefoxwithin** [comment]
> Actually, either works, they do the same thing. Which is cleaner to implement?

It might make more sense for a user of the Library to see the details of the import before or while they import it, similar to how the quest import currently works. Keeping most of this functionality in the same place feels sensible.

Looking further ahead, there's a plan to allow imports to overwrite quests with the same `import_id`. I recall seeing a plan to show detail diffs somewhere on that page as well.

>    If quest does exist, then need some sort of diff view, so users can see what has changed, and then confirm they want to import or not (diff view could be added after, to start we could just indicate "This will overwrite your existing quest, are you sure you want to?")
> Taken from [Sharing Campaigns and Quests from the Library Deck](https://github.com/bytedeck/bytedeck/issues/1546)

Given that, where do you think the detail diffs would best fit? 

Right now, I’m leaning toward including the ability to view the regular quest details and also see the diffs — possibly on a separate page. That way, implementing the detail view now would make adding these planned improvements smoother later.

What are your thoughts?

**Comment by tylerecouture** [comment]
> > I don't want to think about diffs yet! Ok, let's structure it similar to how the quest preview works.
> 
> I'm not saying we add the diffs right now, I'm saying we set up for them by making the view details if that works. Just so that later on we don't have to make new html pages or mess around with unrelated things too much.

Yes!  Please do it!
